astropy/io/registry.py

```diff
@@ -570,14 +570,17 @@ def _is_best_match(class1, class2, format_classes):
     of classes.  It is assumed that (class2 in classes) is True.
     class2 is the the best match if:
       - class1 is class2 => class1 was directly registered.
-      - OR class1 is a subclass of class2 and class1 is not in classes.
-        In this case the subclass will use the parent reader/writer.
+      - OR class1 is a subclass of class2 and class1 is not in classes
+        and class2 is the nearest ancestor of class2 that is in classes.
```

Comment:
Should be `class1 is the nearest ..`.  I think the functional logic is now simpler than what is written, more like 
```
- ``class1`` is a subclass of ``class2`` AND
- ``class2`` is the nearest ancestor of ``class1`` that is in classes 
  (which includes the case that ``class1 is class2``)
```
(The `class1 not in classes` is not really applicable).